pkg/sync/adapter.go

```diff
@@ -0,0 +1,222 @@
+package sync
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/exec"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/pkg/errors"
+)
+
+// New instantiantes a component adapter
+func New(adapterContext common.AdapterContext, client SyncClient) Adapter {
```

Comment:
What's the benefit of an adapter here over just adding functions to a `SyncClient` instance?

I'm fine with either approach here, just curious.

Reply:
Just thought it was cleaner, but yeah i can remove the adapter and perhaps make them functions like the other packages